code/ui/Crosshair.cs

```diff
@@ -117,6 +142,26 @@ public override void Tick()
             BaseCarriable carr = player.ActiveChild as BaseCarriable;
 
             this.Style.Display = carr == null ? DisplayMode.None : DisplayMode.Flex;
+
+            if (ChargeTime > 0)
+            {
+                if (Input.Down(InputButton.Attack1))
+                {
+                    if (ChargeStartTime == 0)
+                    {
+                        ChargeStartTime = Time.Now;
+                    }
+                    float chargePercentage = MathX.Clamp((Time.Now - ChargeStartTime) / ChargeTime, 0, 1) * 100;
```

Comment:
Why using `MathX` here?

Reply:
Could be used inline as well

Reply:
I've explained it above. It can Exceed 1 (100%).

Reply:
No I meant why using `MathX` instead of `Math` here?
Second comment means instead of defining a single used var, you can put this directly into the function as a param.

Reply:
-MathX is just implemented into sbox.
https://wiki.facepunch.com/sbox/Sandbox.MathX
-I merged this variable with function.

Reply:
Yea, I know, but we often run into trouble with sbox-recreated functions that are not as good as cs-sided implementations like the Vector3 system. That's why I suggested usint System's Math.Clamp <https://docs.microsoft.com/en-us/dotnet/api/system.math.clamp?view=net-5.0>, as used in other places in this project as well :)